src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectCodeValidator.groovy

```diff
@@ -0,0 +1,37 @@
+package life.qbic.datamodel.dtos.projectmanagement
+
+import life.qbic.datamodel.dtos.validation.ValidationException
+
+import java.util.function.Consumer
+import java.util.function.Predicate
+import java.util.regex.Pattern
+
+/**
+ * <b>Project Code Validator</b>
+ *
+ * <p>Consumes a project code and throws a {@link ValidationException} in case it is not valid.</p>
+ *
+ * @since 2.12.0
+ */
+class ProjectCodeValidator implements Consumer<String> {
+    private static final Predicate<String> IS_VALID_PROJECT_CODE = Pattern.compile("Q[A-X0-9]{4}").asPredicate()
```

Comment:
Small extensions, to match really single project codes, and not sample codes, entities or even text with a project code contained. A project code starts with a `Q` and ends after four additional alpha-numeric letters.

```suggestion
    private static final Predicate<String> IS_VALID_PROJECT_CODE = Pattern.compile("^Q[A-X0-9]{4}$").asPredicate()
```

Reply:
Nice catch 
![image](https://user-images.githubusercontent.com/11536497/131969831-7de23089-94c6-4715-b474-e93dc0b58d45.png)